Refine the criteria for which tests dump frame load callbacks

darinadler opened the pull request:
#### bf5c2a8d3ba92b79bd68e1e7500c82aac374258d
<pre>
Refine the criteria for which tests dump frame load callbacks
<a href="https://rdar.apple.com/166845643">rdar://166845643</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=304314">https://bugs.webkit.org/show_bug.cgi?id=304314</a>

Reviewed by Alexey Proskuryakov.

* LayoutTests/http/tests/loading/307-after-303-after-post-expected.txt:
* LayoutTests/http/tests/loading/307-after-303-after-post.html:
* LayoutTests/http/tests/loading/authentication-after-redirect-stores-wrong-credentials/authentication-after-redirect-stores-wrong-credentials-expected.txt:
* LayoutTests/http/tests/loading/authentication-after-redirect-stores-wrong-credentials/authentication-after-redirect-stores-wrong-credentials.html:
* LayoutTests/http/tests/loading/bad-scheme-subframe-expected.txt:
* LayoutTests/http/tests/loading/bad-scheme-subframe.html:
* LayoutTests/http/tests/loading/bad-server-subframe-expected.txt:
* LayoutTests/http/tests/loading/bad-server-subframe.html:
* LayoutTests/http/tests/loading/basic-auth-load-URL-with-consecutive-slashes-expected.txt:
* LayoutTests/http/tests/loading/basic-auth-load-URL-with-consecutive-slashes.html:
* LayoutTests/http/tests/loading/basic-auth-remove-credentials-expected.txt:
* LayoutTests/http/tests/loading/basic-auth-remove-credentials.html:
* LayoutTests/http/tests/loading/basic-auth-resend-wrong-credentials-expected.txt:
* LayoutTests/http/tests/loading/basic-auth-resend-wrong-credentials.html:
* LayoutTests/http/tests/loading/basic-credentials-sent-automatically-expected.txt:
* LayoutTests/http/tests/loading/basic-credentials-sent-automatically.html:
* LayoutTests/http/tests/loading/basic-expected.txt:
* LayoutTests/http/tests/loading/basic.html:
* LayoutTests/http/tests/loading/cross-origin-XHR-willLoadRequest-expected.txt:
* LayoutTests/http/tests/loading/cross-origin-XHR-willLoadRequest.html:
* LayoutTests/http/tests/loading/deleted-host-in-resource-load-delegate-callback-expected.txt:
* LayoutTests/http/tests/loading/deleted-host-in-resource-load-delegate-callback.html:
* LayoutTests/http/tests/loading/do-not-preload-css-blocked-by-csp-expected.txt:
* LayoutTests/http/tests/loading/do-not-preload-css-blocked-by-csp.html:
* LayoutTests/http/tests/loading/do-not-preload-script-src-blocked-by-csp-expected.txt:
* LayoutTests/http/tests/loading/do-not-preload-script-src-blocked-by-csp.html:
* LayoutTests/http/tests/loading/empty-subframe-expected.txt:
* LayoutTests/http/tests/loading/empty-subframe.html:
* LayoutTests/http/tests/loading/fire-error-event-empty-404-script-expected.txt:
* LayoutTests/http/tests/loading/fire-error-event-empty-404-script.html:
* LayoutTests/http/tests/loading/fire-error-event-script-no-content-type-expected.txt:
* LayoutTests/http/tests/loading/fire-error-event-script-no-content-type.html:
* LayoutTests/http/tests/loading/hidpi-preload-picture-sizes-expected.txt:
* LayoutTests/http/tests/loading/hidpi-preload-picture-sizes.html:
* LayoutTests/http/tests/loading/load-javascript-after-many-xhrs-expected.txt:
* LayoutTests/http/tests/loading/load-javascript-after-many-xhrs.html:
* LayoutTests/http/tests/loading/location-hash-reload-cycle-expected.txt:
* LayoutTests/http/tests/loading/location-hash-reload-cycle.html:
* LayoutTests/http/tests/loading/main-resource-delegates-on-back-navigation-expected.txt:
* LayoutTests/http/tests/loading/main-resource-delegates-on-back-navigation.html:
* LayoutTests/http/tests/loading/multiple-prioritization-expected.txt:
* LayoutTests/http/tests/loading/multiple-prioritization.html:
* LayoutTests/http/tests/loading/nested_bad_objects-expected.txt:
* LayoutTests/http/tests/loading/oauth-expected.txt:
* LayoutTests/http/tests/loading/oauth.html:
* LayoutTests/http/tests/loading/onload-vs-immediate-refresh-expected.txt:
* LayoutTests/http/tests/loading/pdf-commit-load-callbacks-expected.txt:
* LayoutTests/http/tests/loading/pdf-commit-load-callbacks.html:
* LayoutTests/http/tests/loading/post-in-iframe-with-back-navigation-expected.txt:
* LayoutTests/http/tests/loading/post-in-iframe-with-back-navigation.html:
* LayoutTests/http/tests/loading/preload-append-scan-expected.txt:
* LayoutTests/http/tests/loading/preload-css-import-layer-expected.txt:
* LayoutTests/http/tests/loading/preload-css-import-layer.html:
* LayoutTests/http/tests/loading/preload-css-test-expected.txt:
* LayoutTests/http/tests/loading/preload-css-test.html:
* LayoutTests/http/tests/loading/preload-css-with-csp-nonce-expected.txt:
* LayoutTests/http/tests/loading/preload-css-with-csp-nonce.html:
* LayoutTests/http/tests/loading/preload-ignore-invalid-base-expected.txt:
* LayoutTests/http/tests/loading/preload-img-src-expected.txt:
* LayoutTests/http/tests/loading/preload-img-src.html:
* LayoutTests/http/tests/loading/preload-img-srcset-expected.txt:
* LayoutTests/http/tests/loading/preload-img-srcset-sizes-expected.txt:
* LayoutTests/http/tests/loading/preload-img-srcset-sizes.html:
* LayoutTests/http/tests/loading/preload-img-srcset.html:
* LayoutTests/http/tests/loading/preload-img-test-expected.txt:
* LayoutTests/http/tests/loading/preload-picture-invalid-expected.txt:
* LayoutTests/http/tests/loading/preload-picture-invalid.html:
* LayoutTests/http/tests/loading/preload-picture-nested-expected.txt:
* LayoutTests/http/tests/loading/preload-picture-nested.html:
* LayoutTests/http/tests/loading/preload-picture-sizes-expected.txt:
* LayoutTests/http/tests/loading/preload-picture-sizes.html:
* LayoutTests/http/tests/loading/preload-script-src-expected.txt:
* LayoutTests/http/tests/loading/preload-script-src-with-csp-nonce-expected.txt:
* LayoutTests/http/tests/loading/preload-script-src-with-csp-nonce.html:
* LayoutTests/http/tests/loading/preload-script-src.html:
* LayoutTests/http/tests/loading/preload-script-srcset-ignored-expected.txt:
* LayoutTests/http/tests/loading/preload-script-srcset-ignored.html:
* LayoutTests/http/tests/loading/preload-slow-loading-expected.txt:
* LayoutTests/http/tests/loading/progress-finished-callback-expected.txt:
* LayoutTests/http/tests/loading/progress-finished-callback.html:
* LayoutTests/http/tests/loading/promote-img-preload-priority-expected.txt:
* LayoutTests/http/tests/loading/promote-img-preload-priority.html:
* LayoutTests/http/tests/loading/redirect-methods-expected.txt:
* LayoutTests/http/tests/loading/redirect-methods.html:
* LayoutTests/http/tests/loading/redirect-with-no-location-crash-expected.txt:
* LayoutTests/http/tests/loading/redirect-with-no-location-crash.html:
* LayoutTests/http/tests/loading/remove-child-triggers-parser-expected.txt:
* LayoutTests/http/tests/loading/remove-child-triggers-parser.html:
* LayoutTests/http/tests/loading/server-redirect-for-provisional-load-caching-expected.txt:
* LayoutTests/http/tests/loading/server-redirect-for-provisional-load-caching.html:
* LayoutTests/http/tests/loading/simple-subframe-expected.txt: Copied from LayoutTests/http/tests/loading/empty-subframe-expected.txt.
* LayoutTests/http/tests/loading/simple-subframe.html:
* LayoutTests/http/tests/loading/sizes/preload-image-sizes-2x-expected.txt:
* LayoutTests/http/tests/loading/sizes/preload-image-sizes-2x.html:
* LayoutTests/http/tests/loading/sizes/preload-image-sizes-expected.txt:
* LayoutTests/http/tests/loading/sizes/preload-image-sizes.html:
* LayoutTests/http/tests/loading/slow-parsing-subframe-expected.txt:
* LayoutTests/http/tests/loading/slow-parsing-subframe.html:
* LayoutTests/http/tests/loading/state-object-security-exception-expected.txt:
* LayoutTests/http/tests/loading/state-object-security-exception.html:
* LayoutTests/http/tests/loading/text-content-type-with-binary-extension-expected.txt:
* LayoutTests/http/tests/loading/text-content-type-with-binary-extension.html:
* LayoutTests/http/tests/loading/unfinished-load-back-to-cached-page-callbacks-expected.txt:
* LayoutTests/http/tests/loading/unfinished-load-back-to-cached-page-callbacks.html:
* LayoutTests/http/tests/loading/unfinished-main-resource-back-to-cached-page-callbacks-expected.txt:
* LayoutTests/http/tests/loading/unfinished-main-resource-back-to-cached-page-callbacks.html:
* LayoutTests/http/wpt/loading/redirect-headers-expected.txt:
* LayoutTests/http/wpt/loading/redirect-headers.html:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-css-import-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-css-import-no-quote-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-css-import-no-semicolon-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-css-import-no-space-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-css-import-single-quote-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-link-media-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-template-expected.txt:
* LayoutTests/platform/glib/http/tests/loading/oauth-expected.txt:
* LayoutTests/platform/glib/http/tests/loading/simple-subframe-expected.txt: Removed.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-css-import-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-css-import-no-quote-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-css-import-no-semicolon-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-css-import-no-space-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-css-import-single-quote-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-link-media-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-template-expected.txt:
* LayoutTests/platform/gtk/http/tests/loading/pdf-commit-load-callbacks-expected.txt:
* LayoutTests/platform/ios/http/tests/loading/simple-subframe-expected.txt: Removed.
* LayoutTests/platform/ios/webarchive/loading/cache-expired-subresource-expected.txt:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac-wk1/http/tests/loading/simple-subframe-expected.txt: Added.
* LayoutTests/platform/mac-wk2/http/tests/loading/simple-subframe-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/webarchive/loading/cache-expired-subresource-expected.txt:
* LayoutTests/platform/mac/http/tests/loading/simple-subframe-expected.txt: Removed.
* LayoutTests/platform/mac/webarchive/loading/cache-expired-subresource-expected.txt:
* LayoutTests/platform/win/http/tests/loading/simple-subframe-expected.txt: Removed.
* LayoutTests/platform/win/imported/w3c/web-platform-tests/loading/preloader-css-import-no-quote.tentative-expected.txt:
* LayoutTests/platform/win/imported/w3c/web-platform-tests/loading/preloader-css-import-no-semicolon.tentative-expected.txt:
* LayoutTests/platform/win/imported/w3c/web-platform-tests/loading/preloader-css-import-no-space.tentative-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/authentication-after-redirect-stores-wrong-credentials/authentication-after-redirect-stores-wrong-credentials-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/bad-server-subframe-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/basic-auth-load-URL-with-consecutive-slashes-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/basic-auth-resend-wrong-credentials-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/basic-credentials-sent-automatically-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/oauth-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/redirect-with-no-location-crash-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/server-redirect-for-provisional-load-caching-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/slow-parsing-subframe-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/javascript-url-iframe-crash-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/mainresource-null-mimetype-crash-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/missing-data-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/object-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/test-loading-archive-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/test-loading-archive-subresource-null-mimetype-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/test-loading-top-archive-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/video-in-webarchive-expected.txt:
* LayoutTests/webarchive/loading/cache-expired-subresource.html:
* LayoutTests/webarchive/loading/javascript-url-iframe-crash-expected.txt:
* LayoutTests/webarchive/loading/javascript-url-iframe-crash.html:
* LayoutTests/webarchive/loading/mainresource-null-mimetype-crash-expected.txt:
* LayoutTests/webarchive/loading/mainresource-null-mimetype-crash.html:
* LayoutTests/webarchive/loading/missing-data-expected.txt:
* LayoutTests/webarchive/loading/missing-data.html:
* LayoutTests/webarchive/loading/object-expected.txt:
* LayoutTests/webarchive/loading/object.html:
* LayoutTests/webarchive/loading/test-loading-archive-expected.txt:
* LayoutTests/webarchive/loading/test-loading-archive-subresource-expected.txt:
* LayoutTests/webarchive/loading/test-loading-archive-subresource-null-mimetype-expected.txt:
* LayoutTests/webarchive/loading/test-loading-archive-subresource-null-mimetype.html:
* LayoutTests/webarchive/loading/test-loading-archive-subresource.html:
* LayoutTests/webarchive/loading/test-loading-archive-with-link-expected.txt:
* LayoutTests/webarchive/loading/test-loading-archive-with-link.html:
* LayoutTests/webarchive/loading/test-loading-archive.html:
* LayoutTests/webarchive/loading/test-loading-top-archive-expected.txt:
* LayoutTests/webarchive/loading/test-loading-top-archive.html:
* LayoutTests/webarchive/loading/video-in-webarchive-expected.txt:
* LayoutTests/webarchive/loading/video-in-webarchive.html:
Updated tests to not rely on frame load callbacks being on automatically based on directory name.

* LayoutTests/platform/mac-wk1/TestExpectations: Fixed three expectations that had paths that don&apos;t
match any current tests.

* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(runTest): Removed code to turn on frame load callbacks automatically based on directory name.
(shouldLogFrameLoadDelegates): Deleted.

* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::TestInvocation): Removed code to turn on frame load callbacks automatically based on directory name.

Canonical link: <a href="https://commits.webkit.org/305432@main">https://commits.webkit.org/305432@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eff524babea241e691ef6d7c8552c06aa496146c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/138417 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/10782 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/49827 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/146501 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/91392 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/140291 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/11486 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/10936 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/105906 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/77256 "4 flakes 1 failures") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/b8741657-9737-4fc0-aad8-a139d7357009) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/141364 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/8623 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/124086 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/86751 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/2157cf18-c9fb-4179-8ea4-15d9b6688efb) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/8210 "Passed tests") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/5981 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🛠 wpe-cairo-libwebrtc~~](https://ews-build.webkit.org/#/builders/166/builds/6791 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/117630 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/42285 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/149219 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/10464 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/42842 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/114300 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/10481 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/8849 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/114643 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/29105 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/8280 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/120372 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/65338 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/10512 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/38305 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/10246 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/10451 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/10302 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on LayoutTests/http/wpt/loading/redirect-headers.html:
```diff
@@ -2,6 +2,7 @@
 <script>
 if (window.testRunner) {
     testRunner.dumpAsText();
+    testRunner.dumpFrameLoadCallbacks();
```
I'm not sure why this is desirable. Even though it's not a real WPT test, it's WPT style, and those don't depend on console output.

darinadler replied:
The `redirect-headers-expected.txt` file contains only output that looks like frame load callbacks to me. Given that, it seemed like I needed the call to `dumpFrameLoadCallbacks` to make the test do anything at all. Your comment about WPT style not depending on console output probably means I’m missing something, but I don’t know what!

aproskuryakov replied:
I see, this change makes sense then. I don't think that you are missing anything. Test in `http/wpt` can really do anything, they don't have to be exactly appropriate for WPT upstream. This particular test however is arguably misplaced, given that it doesn't use the WPT harness at all, and only uses the redirect script from `wpt/beacon`.

aproskuryakov commented on LayoutTests/http/tests/loading/preload-slow-loading-expected.txt:
```diff
@@ -1,8 +1,3 @@
-main frame - didCommitLoadForFrame
 non-existant-1 has MIME type text/html
 non-existant-2 has MIME type text/html
-main frame - didFinishDocumentLoadForFrame
```
Is it intentional to not have logging in this test any more?

darinadler replied:
Yes, its was intentional, but perhaps not correct.

I think many of these tests don’t need frame load callback logging to be effective tests, and this is one of those. I didn’t try to be exhaustive on removing the logging when it’s not needed, but I did omit it here. I presume we can do even more of that incrementally after this pull request is landed.

I’m open to the possibility that I am wrong and that this isn’t an effective test without the frame load callback logging.

aproskuryakov replied:
This is a test for bug 41797, and while there is some explanation of what's going on, it seems like fully understanding whether the test is still effective (even before the proposed change) would take effort.

aproskuryakov commented on LayoutTests/http/tests/loading/preload-img-test-expected.txt:
```diff
@@ -1,6 +1,3 @@
-main frame - didStartProvisionalLoadForFrame
-main frame - didCommitLoadForFrame
-main frame - didFinishDocumentLoadForFrame
```
This removes more logging than other tests, and I'm not sure if this test needs logging at all.

darinadler replied:
After considering your comment, I removed the frame load callback logging from this test entirely.

aproskuryakov commented on LayoutTests/http/tests/loading/location-hash-reload-cycle-expected.txt:
```diff
@@ -1,6 +1,3 @@
-main frame - didStartProvisionalLoadForFrame
-main frame - didCommitLoadForFrame
-main frame - didFinishDocumentLoadForFrame
```
This removes more logging than other tests, and I'm not sure if it's needed at all.

darinadler replied:
I think the logging is probably needed. I am basing that guess on the following:
> If this test fails when run in DumpRenderTree, the FrameLoader callback output will contain willPerformClientRedirectToURL callbacks.

aproskuryakov reviewed (approved):
This is great! For some reason, I was always assuming that this couldn't be in JS because setting it in JS is too late to catch important loading events. But looking at the modified results, I also don't see anything important.

There is still a failing test.

darinadler committed:
Refine the criteria for which tests dump frame load callbacks
rdar://166845643
https://bugs.webkit.org/show_bug.cgi?id=304314

Reviewed by Alexey Proskuryakov.

* LayoutTests/http/tests/loading/307-after-303-after-post-expected.txt:
* LayoutTests/http/tests/loading/307-after-303-after-post.html:
* LayoutTests/http/tests/loading/authentication-after-redirect-stores-wrong-credentials/authentication-after-redirect-stores-wrong-credentials-expected.txt:
* LayoutTests/http/tests/loading/authentication-after-redirect-stores-wrong-credentials/authentication-after-redirect-stores-wrong-credentials.html:
* LayoutTests/http/tests/loading/bad-scheme-subframe-expected.txt:
* LayoutTests/http/tests/loading/bad-scheme-subframe.html:
* LayoutTests/http/tests/loading/bad-server-subframe-expected.txt:
* LayoutTests/http/tests/loading/bad-server-subframe.html:
* LayoutTests/http/tests/loading/basic-auth-load-URL-with-consecutive-slashes-expected.txt:
* LayoutTests/http/tests/loading/basic-auth-load-URL-with-consecutive-slashes.html:
* LayoutTests/http/tests/loading/basic-auth-remove-credentials-expected.txt:
* LayoutTests/http/tests/loading/basic-auth-remove-credentials.html:
* LayoutTests/http/tests/loading/basic-auth-resend-wrong-credentials-expected.txt:
* LayoutTests/http/tests/loading/basic-auth-resend-wrong-credentials.html:
* LayoutTests/http/tests/loading/basic-credentials-sent-automatically-expected.txt:
* LayoutTests/http/tests/loading/basic-credentials-sent-automatically.html:
* LayoutTests/http/tests/loading/basic-expected.txt:
* LayoutTests/http/tests/loading/basic.html:
* LayoutTests/http/tests/loading/cross-origin-XHR-willLoadRequest-expected.txt:
* LayoutTests/http/tests/loading/cross-origin-XHR-willLoadRequest.html:
* LayoutTests/http/tests/loading/deleted-host-in-resource-load-delegate-callback-expected.txt:
* LayoutTests/http/tests/loading/deleted-host-in-resource-load-delegate-callback.html:
* LayoutTests/http/tests/loading/do-not-preload-css-blocked-by-csp-expected.txt:
* LayoutTests/http/tests/loading/do-not-preload-css-blocked-by-csp.html:
* LayoutTests/http/tests/loading/do-not-preload-script-src-blocked-by-csp-expected.txt:
* LayoutTests/http/tests/loading/do-not-preload-script-src-blocked-by-csp.html:
* LayoutTests/http/tests/loading/empty-subframe-expected.txt:
* LayoutTests/http/tests/loading/empty-subframe.html:
* LayoutTests/http/tests/loading/fire-error-event-empty-404-script-expected.txt:
* LayoutTests/http/tests/loading/fire-error-event-empty-404-script.html:
* LayoutTests/http/tests/loading/fire-error-event-script-no-content-type-expected.txt:
* LayoutTests/http/tests/loading/fire-error-event-script-no-content-type.html:
* LayoutTests/http/tests/loading/hidpi-preload-picture-sizes-expected.txt:
* LayoutTests/http/tests/loading/hidpi-preload-picture-sizes.html:
* LayoutTests/http/tests/loading/load-javascript-after-many-xhrs-expected.txt:
* LayoutTests/http/tests/loading/load-javascript-after-many-xhrs.html:
* LayoutTests/http/tests/loading/location-hash-reload-cycle-expected.txt:
* LayoutTests/http/tests/loading/location-hash-reload-cycle.html:
* LayoutTests/http/tests/loading/main-resource-delegates-on-back-navigation-expected.txt:
* LayoutTests/http/tests/loading/main-resource-delegates-on-back-navigation.html:
* LayoutTests/http/tests/loading/multiple-prioritization-expected.txt:
* LayoutTests/http/tests/loading/multiple-prioritization.html:
* LayoutTests/http/tests/loading/nested_bad_objects-expected.txt:
* LayoutTests/http/tests/loading/oauth-expected.txt:
* LayoutTests/http/tests/loading/oauth.html:
* LayoutTests/http/tests/loading/onload-vs-immediate-refresh-expected.txt:
* LayoutTests/http/tests/loading/pdf-commit-load-callbacks-expected.txt:
* LayoutTests/http/tests/loading/pdf-commit-load-callbacks.html:
* LayoutTests/http/tests/loading/post-in-iframe-with-back-navigation-expected.txt:
* LayoutTests/http/tests/loading/post-in-iframe-with-back-navigation.html:
* LayoutTests/http/tests/loading/preload-append-scan-expected.txt:
* LayoutTests/http/tests/loading/preload-css-import-layer-expected.txt:
* LayoutTests/http/tests/loading/preload-css-import-layer.html:
* LayoutTests/http/tests/loading/preload-css-test-expected.txt:
* LayoutTests/http/tests/loading/preload-css-test.html:
* LayoutTests/http/tests/loading/preload-css-with-csp-nonce-expected.txt:
* LayoutTests/http/tests/loading/preload-css-with-csp-nonce.html:
* LayoutTests/http/tests/loading/preload-ignore-invalid-base-expected.txt:
* LayoutTests/http/tests/loading/preload-img-src-expected.txt:
* LayoutTests/http/tests/loading/preload-img-src.html:
* LayoutTests/http/tests/loading/preload-img-srcset-expected.txt:
* LayoutTests/http/tests/loading/preload-img-srcset-sizes-expected.txt:
* LayoutTests/http/tests/loading/preload-img-srcset-sizes.html:
* LayoutTests/http/tests/loading/preload-img-srcset.html:
* LayoutTests/http/tests/loading/preload-img-test-expected.txt:
* LayoutTests/http/tests/loading/preload-picture-invalid-expected.txt:
* LayoutTests/http/tests/loading/preload-picture-invalid.html:
* LayoutTests/http/tests/loading/preload-picture-nested-expected.txt:
* LayoutTests/http/tests/loading/preload-picture-nested.html:
* LayoutTests/http/tests/loading/preload-picture-sizes-expected.txt:
* LayoutTests/http/tests/loading/preload-picture-sizes.html:
* LayoutTests/http/tests/loading/preload-script-src-expected.txt:
* LayoutTests/http/tests/loading/preload-script-src-with-csp-nonce-expected.txt:
* LayoutTests/http/tests/loading/preload-script-src-with-csp-nonce.html:
* LayoutTests/http/tests/loading/preload-script-src.html:
* LayoutTests/http/tests/loading/preload-script-srcset-ignored-expected.txt:
* LayoutTests/http/tests/loading/preload-script-srcset-ignored.html:
* LayoutTests/http/tests/loading/preload-slow-loading-expected.txt:
* LayoutTests/http/tests/loading/progress-finished-callback-expected.txt:
* LayoutTests/http/tests/loading/progress-finished-callback.html:
* LayoutTests/http/tests/loading/promote-img-preload-priority-expected.txt:
* LayoutTests/http/tests/loading/promote-img-preload-priority.html:
* LayoutTests/http/tests/loading/redirect-methods-expected.txt:
* LayoutTests/http/tests/loading/redirect-methods.html:
* LayoutTests/http/tests/loading/redirect-with-no-location-crash-expected.txt:
* LayoutTests/http/tests/loading/redirect-with-no-location-crash.html:
* LayoutTests/http/tests/loading/remove-child-triggers-parser-expected.txt:
* LayoutTests/http/tests/loading/remove-child-triggers-parser.html:
* LayoutTests/http/tests/loading/server-redirect-for-provisional-load-caching-expected.txt:
* LayoutTests/http/tests/loading/server-redirect-for-provisional-load-caching.html:
* LayoutTests/http/tests/loading/simple-subframe-expected.txt: Copied from LayoutTests/http/tests/loading/empty-subframe-expected.txt.
* LayoutTests/http/tests/loading/simple-subframe.html:
* LayoutTests/http/tests/loading/sizes/preload-image-sizes-2x-expected.txt:
* LayoutTests/http/tests/loading/sizes/preload-image-sizes-2x.html:
* LayoutTests/http/tests/loading/sizes/preload-image-sizes-expected.txt:
* LayoutTests/http/tests/loading/sizes/preload-image-sizes.html:
* LayoutTests/http/tests/loading/slow-parsing-subframe-expected.txt:
* LayoutTests/http/tests/loading/slow-parsing-subframe.html:
* LayoutTests/http/tests/loading/state-object-security-exception-expected.txt:
* LayoutTests/http/tests/loading/state-object-security-exception.html:
* LayoutTests/http/tests/loading/text-content-type-with-binary-extension-expected.txt:
* LayoutTests/http/tests/loading/text-content-type-with-binary-extension.html:
* LayoutTests/http/tests/loading/unfinished-load-back-to-cached-page-callbacks-expected.txt:
* LayoutTests/http/tests/loading/unfinished-load-back-to-cached-page-callbacks.html:
* LayoutTests/http/tests/loading/unfinished-main-resource-back-to-cached-page-callbacks-expected.txt:
* LayoutTests/http/tests/loading/unfinished-main-resource-back-to-cached-page-callbacks.html:
* LayoutTests/http/wpt/loading/redirect-headers-expected.txt:
* LayoutTests/http/wpt/loading/redirect-headers.html:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-css-import-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-css-import-no-quote-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-css-import-no-semicolon-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-css-import-no-space-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-css-import-single-quote-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-link-media-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/loading/preloader-template-expected.txt:
* LayoutTests/platform/glib/http/tests/loading/oauth-expected.txt:
* LayoutTests/platform/glib/http/tests/loading/simple-subframe-expected.txt: Removed.
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-css-import-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-css-import-no-quote-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-css-import-no-semicolon-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-css-import-no-space-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-css-import-single-quote-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-link-media-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/loading/preloader-template-expected.txt:
* LayoutTests/platform/gtk/http/tests/loading/pdf-commit-load-callbacks-expected.txt:
* LayoutTests/platform/ios/http/tests/loading/simple-subframe-expected.txt: Removed.
* LayoutTests/platform/ios/webarchive/loading/cache-expired-subresource-expected.txt:
* LayoutTests/platform/mac-wk1/TestExpectations:
* LayoutTests/platform/mac-wk1/http/tests/loading/simple-subframe-expected.txt: Added.
* LayoutTests/platform/mac-wk2/http/tests/loading/simple-subframe-expected.txt: Removed.
* LayoutTests/platform/mac-wk2/webarchive/loading/cache-expired-subresource-expected.txt:
* LayoutTests/platform/mac/http/tests/loading/simple-subframe-expected.txt: Removed.
* LayoutTests/platform/mac/webarchive/loading/cache-expired-subresource-expected.txt:
* LayoutTests/platform/win/http/tests/loading/simple-subframe-expected.txt: Removed.
* LayoutTests/platform/win/imported/w3c/web-platform-tests/loading/preloader-css-import-no-quote.tentative-expected.txt:
* LayoutTests/platform/win/imported/w3c/web-platform-tests/loading/preloader-css-import-no-semicolon.tentative-expected.txt:
* LayoutTests/platform/win/imported/w3c/web-platform-tests/loading/preloader-css-import-no-space.tentative-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/authentication-after-redirect-stores-wrong-credentials/authentication-after-redirect-stores-wrong-credentials-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/bad-server-subframe-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/basic-auth-load-URL-with-consecutive-slashes-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/basic-auth-resend-wrong-credentials-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/basic-credentials-sent-automatically-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/oauth-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/redirect-with-no-location-crash-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/server-redirect-for-provisional-load-caching-expected.txt:
* LayoutTests/platform/wk2/http/tests/loading/slow-parsing-subframe-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/javascript-url-iframe-crash-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/mainresource-null-mimetype-crash-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/missing-data-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/object-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/test-loading-archive-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/test-loading-archive-subresource-null-mimetype-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/test-loading-top-archive-expected.txt:
* LayoutTests/platform/wk2/webarchive/loading/video-in-webarchive-expected.txt:
* LayoutTests/webarchive/loading/cache-expired-subresource.html:
* LayoutTests/webarchive/loading/javascript-url-iframe-crash-expected.txt:
* LayoutTests/webarchive/loading/javascript-url-iframe-crash.html:
* LayoutTests/webarchive/loading/mainresource-null-mimetype-crash-expected.txt:
* LayoutTests/webarchive/loading/mainresource-null-mimetype-crash.html:
* LayoutTests/webarchive/loading/missing-data-expected.txt:
* LayoutTests/webarchive/loading/missing-data.html:
* LayoutTests/webarchive/loading/object-expected.txt:
* LayoutTests/webarchive/loading/object.html:
* LayoutTests/webarchive/loading/test-loading-archive-expected.txt:
* LayoutTests/webarchive/loading/test-loading-archive-subresource-expected.txt:
* LayoutTests/webarchive/loading/test-loading-archive-subresource-null-mimetype-expected.txt:
* LayoutTests/webarchive/loading/test-loading-archive-subresource-null-mimetype.html:
* LayoutTests/webarchive/loading/test-loading-archive-subresource.html:
* LayoutTests/webarchive/loading/test-loading-archive-with-link-expected.txt:
* LayoutTests/webarchive/loading/test-loading-archive-with-link.html:
* LayoutTests/webarchive/loading/test-loading-archive.html:
* LayoutTests/webarchive/loading/test-loading-top-archive-expected.txt:
* LayoutTests/webarchive/loading/test-loading-top-archive.html:
* LayoutTests/webarchive/loading/video-in-webarchive-expected.txt:
* LayoutTests/webarchive/loading/video-in-webarchive.html:
Updated tests to not rely on frame load callbacks being on automatically based on directory name.

* LayoutTests/platform/mac-wk1/TestExpectations: Fixed three expectations that had paths that don't
match any current tests.

* Tools/DumpRenderTree/mac/DumpRenderTree.mm:
(runTest): Removed code to turn on frame load callbacks automatically based on directory name.
(shouldLogFrameLoadDelegates): Deleted.

* Tools/WebKitTestRunner/TestInvocation.cpp:
(WTR::TestInvocation::TestInvocation): Removed code to turn on frame load callbacks automatically based on directory name.

Canonical link: https://commits.webkit.org/305432@main
